app/services/caesar_importer.rb

```diff
@@ -0,0 +1,77 @@
+class CaesarImporter
+  attr_reader :reduction_id, :reducible, :data, :subject_info
+
+  def initialize(**args)
+    @reduction_id = args[:id]
+    @reducible = args[:reducible]
+    @data = args[:data]
+    @subject_info = args[:subject]
+  end
+
+  def process
+    generate_parent_resources
+
+    # If a transcription with this id already exists, an exception will be raised
+    new_transcription = Transcription.new(transcription_create_attrs)
+    new_transcription.workflow = workflow
+    new_transcription.save!
+  end
+
+  private
+
+  def workflow
+    Workflow.find_by(id: reducible[:id])
+  end
+
+  def generate_parent_resources
+    # If the workflow exists, the project does too
+    unless workflow
+      response = pull_wf_and_project
+      existing_project = Project.find_by(id: response[:project][:id])
+
+      if existing_project
+        # Project exists, create workflow
+        new_wf = Workflow.create(
```

Comment:
I think you should use create! here to raise as we don't test the truthy value if it doesn't create.

Reply:
👍